kornia/augmentation/param_gen.py

```diff
@@ -258,12 +258,46 @@ def _random_rotation_gen(batch_size: int, degrees: FloatUnionType) -> Dict[str,
     return params
 
 
+def _compose_affine_matrix_3x3(translations: torch.Tensor,
+                               center: torch.Tensor,
+                               scale: torch.Tensor,
+                               angle: torch.Tensor,
+                               sx: Optional[torch.Tensor],
+                               sy: Optional[torch.Tensor]) -> torch.Tensor:
+    r"""Composes affine matrix Bx3x3 from the components
+    Returns:
+        torch.Tensor: params to be passed to the affine transformation.
+    """
+    if sx is not None:
+        x, y = torch.split(center, 1, dim=-1)
+        x = x.view(-1)
+        y = y.view(-1)
+        sx_tan = torch.tan(sx)  # type: ignore
+        sy_tan = torch.tan(sy)  # type: ignore
+        zeros = torch.zeros_like(sx)  # type: ignore
+        ones = torch.ones_like(sx)  # type: ignore
+        shear_mat = torch.stack([ones,   -sx_tan,                 sx_tan * x,  # type: ignore   # noqa: E241
+                                 -sy_tan, ones + sx_tan * sy_tan, sy_tan * (-sx_tan * x + y)],  # noqa: E241
+                                dim=-1).view(-1, 2, 3)
+        shear_mat = torch.nn.functional.pad(shear_mat, [0, 0, 0, 1], value=0.)
+        shear_mat[..., -1, -1] += 1.0
+    transform: torch.Tensor = get_rotation_matrix2d(center, -angle, scale)
+    transform[..., 2] += translations  # tx/ty
+    # pad transform to get Bx3x3
+    transform_h = torch.nn.functional.pad(transform, [0, 0, 0, 1], value=0.)
+    transform_h[..., -1, -1] += 1.0
+    if sx is not None:
+        transform_h = torch.bmm(transform_h, shear_mat)
```

Comment:
just curious, why not `torch.matmul` or `@`

Reply:
Just habit :) I associate matmul with mm, i.e. non-batched and @ didn't work in python 2))